doc/api/http2.md

```diff
@@ -0,0 +1,1704 @@
+
+#### http2session.setTimeout(msecs, callback)
+<!-- YAML
+added: REPLACEME
+-->
+
+* `msecs` {number}
+* `callback` {Function}
+* Returns: {undefined}
+
+Used to set a callback function that is called when there is no activity on
+the `Http2Session` after `msecs` milliseconds. The given `callback` is
+registered as a listener on the `'timeout'` event.
+
+#### http2session.shutdown(options[, callback])
+<!-- YAML
+added: REPLACEME
+-->
+
+* `options` {Object}
+  * `graceful` {boolean} `true` to attempt a polite shutdown of the
+    `Http2Session`.
+  * `errorCode` {number} The HTTP/2 [error code][] to return. Note that this is
+    *not* the same thing as an HTTP Response Status Code. Defaults to `0x00`
+    (No Error).
+  * `lastStreamID` {number} The Stream ID of the last successfully processed
+    `Http2Stream` on this `Http2Session`.
+  * `opaqueData` {Buffer} A `Buffer` instance containing arbitrary additional
+    data to send to the peer upon disconnection. This is used, typically, to
+    provide additional data for debugging failures, if necessary.
+* `callback` {Function} A callback that is invoked after the session shutdown
+  has been completed.
+* Returns: {undefined}
+
+Attempts to shutdown this `Http2Session` using HTTP/2 defined procedures.
+If specified, the given `callback` function will be invoked once the shutdown
+process has completed.
+
+Note that calling `http2session.shutdown()` does *not* destroy the session or
+tear down the `Socket` connection. It merely prompts both sessions to begin
+preparing to cease activity.
+
+During a "graceful" shutdown, the session will first send a `GOAWAY` frame to
+the connected peer identifying the last processed stream as 2<sup>32</sup>-1.
+Then, on the next tick of the event loop, a second `GOAWAY` frame identifying
+the most recently processed stream identifier is sent. This process allows the
+remote peer to begin preparing for the connection to be terminated.
+
+```js
+session.shutdown({
+  graceful: true,
+  opaqueData: Buffer.from('add some debugging data here')
+}, () => session.destroy());
+```
+
+#### http2session.socket
+<!-- YAML
+added: REPLACEME
+-->
+
+* Value: {net.Socket|tls.TLSSocket}
+
+A reference to the [`net.Socket`][] or [`tls.TLSSocket`][] to which this
+`Http2Session` instance is bound.
+
+*Note*: It is not recommended for user code to interact directly with a
+`Socket` bound to an `Http2Session`. See [Http2Session and Sockets][] for
+details.
+
+#### http2session.state
+<!-- YAML
+added: REPLACEME
+-->
+
+* Value: {Object}
+  * `effectiveLocalWindowSize` {number}
+  * `effectiveRecvDataLength` {number}
+  * `nextStreamID` {number}
+  * `localWindowSize` {number}
+  * `lastProcStreamID` {number}
+  * `remoteWindowSize` {number}
+  * `outboundQueueSize` {number}
+  * `deflateDynamicTableSize` {number}
+  * `inflateDynamicTableSize` {number}
+
+An object describing the current status of this `Http2Session`.
+
+#### http2session.priority(stream, options)
```

Comment:
Currently, yes, but eventually this will support the ability to pass in a numeric stream identifier rather than the `Http2Stream` object. The HTTP/2 spec allows priorities and rst_stream frames to be sent independently, even without a stream having been previously established.